src/overmap.cpp

```diff
@@ -212,10 +212,10 @@ bool overmap_special_id::is_valid() const
 }
 
 city::city( const point &P, int const S )
-    : pos( P )
-    , size( S )
-    , name( Name::get( nameIsTownName ) )
 {
+        pos = P;
+        size = S;
+        name = RandomName::getInstance()->getRandomName(NAME_IS_TOWN_NAME);
```

Comment:
Something like this:
```suggestion
        name = Names::get_random(NAME_IS_TOWN_NAME);
```
Is much shorter, but provides the same information to the reader. And because the context and the name of the enumeration value make clear it's a name, one could even write it as `name = get_random(NAME_IS_TOWN_NAME);`.

Reply:
### What is the advantage of using a singleton class instead of a namespace?

Mention several advantages [here](https://github.com/CleverRaven/Cataclysm-DDA/pull/31413#discussion_r293824726) of using the _Singleton_ pattern

##### Overview

1. Global access point
2. Only instance

###### However, I must say something before I begin to explain why.

Before starting with the refactoring of the code contained within a namespace, I spent days (literally) trying to understand how the function `get` to choose a name from the {flag}, without fear of equivocation is the most intricate code and difficult to understand that I have come to see, so according to what was observed and said by the little documentation, I decided to rewrite the code from scratch for the generation of random names.

I am a person who thinks that simplicity is much better than complexity, what Ockham said.

> (Occam's razor: "The simplest solution is most likely the right one.").

Besides that my opinion considers object-oriented programming as my saviour and my tool to solve these problems.

------

This new class had two characteristics that made it worthy of being of the _Singleton_ type:

1. Global access point
2. Only instance

The first point is by itself explanatory, the second needs deeper reasoning.

The namespace allowed the use of the API to generate names without the need of instacies (we can say, functional programming), and in part it wouldn't have been a problem if the code had been manageable, but it wasn't, it was intricate, it was difficult to understand and much more to debug, it screamed for a change.

I started from the premise that the names were contained within a .json file, and that an Enum was needed to determine the type of name desired.

When I made the change from functional programming to object-oriented programming, I came across the problems of the instances, nothing prevented the existence of two, three, hundred or thousand instances of the class, for me that represented a problem, not only because you did not need more than two instances, since each time you were going to invoke (create or instantiate) a new object had to store the string within vectors, this consumes time and space, is something that could not allow, a good engineer does not allow that.

I used the pattern looking to solve an inadvertent problem, the instance of more than one object of the RandomName type.

Reply:
### The usage is now even more verbose:

There are several ways to solve this problem, from name normalization to the use of member variables or local variables.

In my opinion, nothing serious.

------ 

_This has been made possible by Deepl._